flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -291,6 +297,7 @@ public Task(
 		TaskManagerActions taskManagerActions,
 		InputSplitProvider inputSplitProvider,
 		CheckpointResponder checkpointResponder,
+		OperatorEventSender operatorCoordinatorEventGateway,
```

Comment:
I was thinking about that as well. The reason is that each layer adds information. Within the operator, you just throw out events. In the  StreamTask, this gets augmented with Operator ID and in the Task with Task Execution Attempt Id. I wanted to avoid that the operator has to think about the context it runs in, hence these layers.

I can try and align the names, at lease `OperatorEventGateway` --> `TaskOperatorEventGateway` --> `TaskExecutorOperatorEventGateway`.

What do you think?

Reply:
Renaming would be helpful. It seems we don't need a `TaskExecutorEventGateway` given there are only two tiers of information to fill in.

BTW, not important, but do we want to have a symmetric pattern for the `TaskExecutorGateway` as well? Currently the `sendOperatorEvent()` method is defined in the `TaskExecutorGateway` directly while the `JobMasterGateway` extends the `OperatorCoordinatorEventGateway`.

Reply:
Have renamed, added better JavaDocs, and introduced a symmetric interface on the JM side.